File: gym/type.py
from typing import List, Union
from dataclasses import dataclass

@dataclass
class Player:
    number: str
    opta_id: str
    player_id: str
    speed: float
    xyz: List[float]

@dataclass
class Ball:
    speed: float
    xyz: List[float]

@dataclass
class Frame:
    away_players: List[Player]
    ball: Ball
    frame_idx: int
    game_clock: float
    home_players: List[Player]
    period: int
    wall_clock: int

@dataclass
class Signal:
    end_frame_idx: int
    end_wall_clock: int
    number: int
    start_frame_idx: int
    start_wall_clock: int

@dataclass
class GamePacket:
    league: str
    game_id: str
    feed_name: str
    message_id: str
    data: List[Union[Frame, Signal]]

File: benchmark/type.go
package main

type producedFrame struct {
	Period      int     `json:"period"`
	FrameIdx    int     `json:"frameIdx"`
	GameClock   float64 `json:"gameClock"`
	WallClock   int64   `json:"wallClock"`
	HomePlayers []struct {
		PlayerId string    `json:"playerId"`
		Number   int       `json:"number"`
		Xyz      []float64 `json:"xyz"`
		Speed    float64   `json:"speed"`
		OptaId   string    `json:"optaId"`
	} `json:"homePlayers"`
	AwayPlayers []struct {
		PlayerId string    `json:"playerId"`
		Number   int       `json:"number"`
		Xyz      []float64 `json:"xyz"`
		Speed    float64   `json:"speed"`
		OptaId   string    `json:"optaId"`
	} `json:"awayPlayers"`
	Ball struct {
		Xyz   []float64 `json:"xyz"`
		Speed float64   `json:"speed"`
	} `json:"ball"`
	Live      bool   `json:"live"`
	LastTouch string `json:"lastTouch"`
}
